### PR TITLE
Fix CI with git diff

### PR DESCRIPTION
**Pull request by roidelapluie**
There seems to be a regression in the way git diff works when 2 path are given

Signed-off-by: Julien Pivotto <roidelapluie@o11y.eu>

### CODE REVIEW

File: Makefile.common [hunk 1]
```diff
@@ -190,7 +190,8 @@ common-staticcheck: lint
 common-unused:
 	@echo ">> running check for unused/missing packages in go.mod"
 	$(GO) mod tidy
-	@git diff --exit-code -- go.sum go.mod
+	@git diff --exit-code -- go.sum
```
Comment (bboreham) [inline Makefile.common:193]:
```suggestion
	@git diff @ --exit-code -- go.sum
```

Comment (beorn7) [reply Makefile.common:193]: Isn't that the default behavior anyway?

**Comment by beorn7** [comment]
> warning: Not a git repository. Use --no-index to compare two paths outside a working tree

Where do you see this warning? And why? Aren't we in a git repository?

**Comment by bboreham** [comment]
In CI. 
https://github.com/prometheus/prometheus/actions/runs/4123425393/jobs/7121429824

Look at some recent PRs which are all showing CI failures.

**Comment by roidelapluie** [comment]
fixed in golang builder.

**Comment by bboreham** [comment]
What was the issue?  I didn’t understand the fix.

**Pull request closed without merging**